scanamo/src/main/scala/org/scanamo/query/ConditionExpression.scala

```diff
@@ -13,12 +13,6 @@ case class ConditionalOperation[V, T](tableName: String, t: T)(
   implicit state: ConditionExpression[T],
   format: DynamoFormat[V]
 ) {
-  def put(item: V): ScanamoOps[Either[ConditionalCheckFailedException, PutItemResult]] = {
```

Comment:
If you get rid of this, it won't be possible to do conditional puts anymore. I think this method should stay, but the return type changed according to your changes:
```
def put(item: V): ScanamoOps[ScanamoResult[PutItemResult]] = ScanamoOps.put(
  ScanamoPutRequest(tableName, format.write(item), Some(state.apply(t)(None))
)
```